### DISCUSSION TITLE
Allow saving form field changes in PDFs

### DISCUSSION THREAD

**Discussion by LokiLuciferase** [TriliumNext/Trilium#4616] [Ideas]
It would be really cool if changes to PDF form fields made via the internal PDF viewer would be saved by Trilium. At least when hitting the "save revision" button. I keep my pen and paper character sheets in Trilium, but I always need to download at the start of the session and upload the final version afterwards.

**Comment by zadam** [comment]
This is a limitation of the PDF viewer built-in the electron. I don't think it can save annotations.

**Reply by LokiLuciferase** [reply]
Thanks for looking into it! Seems like using the "Print to file" utility of the PDF viewer does save changes to annotations, though I acknowledge that's a hacky solution and I'm not positive the printing process doesn't change other things about the file as well.

edit: looks like selecting "Download -> with your changes" also preserves those annotations (probably does the same thing internally but certainly sounds less lossy? :sweat_smile: )

edit2: I understand now that what I was seeing here are the controls of the built-in PDF viewer of my browser which may not work the same way in all browsers or in electron itself, my bad for my confusion.
